enginetest/queries/query_plans.go

```diff
@@ -910,6 +910,28 @@ var PlanTests = []QueryPlanTest{
 			"     └─ columns: [a]\n" +
 			"",
 	},
+	{
```

Comment:
can we add some tests 1) with this transform preformed inside subqueries, making sure they stay cached afterwards (`select * from xy join (select count(*) as u, 1 as v from ...) uv on x = u`, and 2)) this transform performed inside subquery expressions, with semi/anti joins still being generated afterwards (` select ... where x in/not in (select y from (select count(*) as u, 1 as v from ...) uv)`, `select ... where x =/!= (...)`, `select * from xy where exists (select y from (select count(*) as u, 1 as v from ...) uv) where x = y)`)

Reply:
and idk if this is valid, but can the projection be a subquery expression? `select count(*), (select * from uv where u = x) from xy group by x`